spec.bs

```diff
@@ -810,6 +810,68 @@ To <dfn>compute protocol matches a special scheme flag</dfn> given a [=construct
   1. If the result of running [=protocol component matches a special scheme=] given |protocol component| is true, then set |parser|'s [=constructor string parser/protocol matches a special scheme flag=] to true.
 </div>
 
+<h2 id=urlpatternlist-class>The {{URLPatternList}} class </h2>
+
+<xmp class="idl">
+[Exposed=(Window,Worker)]
+interface URLPatternList {
+  constructor(sequence<URLPattern> patterns);
+
+  boolean test(optional URLPatternInput input = {}, optional USVString baseURL);
+
+  URLPatternListResult? exec(optional URLPatternInput input = {}, optional USVString baseURL);
+};
+
+dictionary URLPatternListResult : URLPatternResult {
+  URLPattern pattern;
```

Comment:
I don't know how it works internally but shouldn't it contain the match result as well? Since the point is to pick a pattern to match and it's likely doing some work behind the scenes

Reply:
It does — the `: URLPatternResult` part means this dictionary extends [URLPatternResult](https://wicg.github.io/urlpattern/#dictdef-urlpatternresult).

Reply:
Aaah i didn’t see it. (Not familiar with this spec language 😂)